evalml/tests/component_tests/decomposer_tests/test_polynomial_decomposer.py

```diff
@@ -398,19 +399,59 @@ def test_polynomial_decomposer_uses_time_index(
             assert isinstance(y.index, pd.DatetimeIndex)
 
 
+def generate_seasonal_data(
```

Comment:
This would fit well as a fixture in `conftest.py`!

Reply:
So be it.